krm-functions/lib/nad/v1/nad.go

```diff
@@ -0,0 +1,366 @@
+/*
+Copyright 2023 Nephio.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	nadv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	"github.com/nephio-project/nephio/krm-functions/lib/kubeobject"
+)
+
+const (
+	errKubeObjectNotInitialized = "KubeObject not initialized"
+	CniVersion                  = "0.3.1"
+	ModeBridge                  = "bridge"
+	ModeL2                      = "l2"
+	NadType                     = "static"
+	TuningType                  = "tuning"
+)
+
+var (
+	ConfigType = []string{"spec", "config"}
+)
+
+type NadConfig struct {
+	CniVersion string          `json:"cniVersion,omitempty"`
+	Vlan       int             `json:"vlan,omitempty"`
+	Plugins    []PluginCniType `json:"plugins,omitempty"`
+}
+
+type PluginCniType struct {
+	Type         string       `json:"type,omitempty"`
+	Capabilities Capabilities `json:"capabilities,omitempty"`
+	Master       string       `json:"master,omitempty"`
+	Mode         string       `json:"mode,omitempty"`
+	Ipam         Ipam         `json:"ipam,omitempty"`
+}
+
+type Capabilities struct {
+	Ips bool `json:"ips,omitempty"`
+	Mac bool `json:"mac,omitempty"`
+}
+
+type Ipam struct {
+	Type      string      `json:"type,omitempty"`
+	Addresses []Addresses `json:"addresses,omitempty"`
+}
+
+type Addresses struct {
+	Address string `json:"address,omitempty"`
+	Gateway string `json:"gateway,omitempty"`
+}
+
+type CniSpecType int64
+
+const (
+	OtherType CniSpecType = iota // 0
+	VlanAllocOnly
+	IpVlanType
+	SriovType
+	MacVlanType
```

Comment:
Not sure what the test coverage goal for the project is but I am getting just under 70%. 
Still learning the tools so may not be an accurate reflection.

Reply:
Got it over 80%, the rest are simply struct and const definitions..